### DISCUSSION TITLE
RotatingFileHandler for the Logging library

### DISCUSSION THREAD

**Discussion by PolarGoose** [micropython/micropython#15416] [Libraries & Drivers]
I have created a RotatingFileHandler for the Micropython Logging library: [MicroPython-Logging-RotatingFileHandler](https://github.com/PolarGoose/MicroPython-Logging-RotatingFileHandler)
It is identical to the [RotatingFileHandler from CPython](https://docs.python.org/3/library/logging.handlers.html#logging.handlers.RotatingFileHandler)

**Comment by davefes** [comment]
I get a 404

**Reply by scruss** [reply]
Likewise.

OP: you seem to be commenting on the older [pycopy-lib/logging](https://github.com/pfalcon/pycopy-lib/tree/master) repo, and not the current [micropython-lib/python-stdlib/logging/](https://github.com/micropython/micropython-lib/tree/master/python-stdlib/logging) one

**Reply by PolarGoose** [reply]
@davefes, @scruss,
Sorry for the inconvenience. I have fixed the link.
Thank you for your feedback.

> OP: you seem to be commenting on the older [pycopy-lib/logging](https://github.com/pfalcon/pycopy-lib/tree/master) repo, and not the current [micropython-lib/python-stdlib/logging/](https://github.com/micropython/micropython-lib/tree/master/python-stdlib/logging) one

What do you mean?

**Reply by scruss** [reply]
for example, this issue you raised: [\[logging\] sys.exc_info() function doesn't exist on ESP32 · Issue #88 · pfalcon/pycopy-lib](https://github.com/pfalcon/pycopy-lib/issues/88)

pypi has the wrong information for who owns MicroPython projects. In fact, it's fair to say that installing almost anything from pypi for MicroPython will do it wrong. I don't know why Thonny looks up packages there at all.

**Reply by PolarGoose** [reply]
@scruss,
Thank you for the explanation. Indeed, I didn't know that [micropython-lib/python-stdlib/logging](https://github.com/micropython/micropython-lib/tree/master/python-stdlib/logging) existed.

**Reply by Josverl** [reply]
>  it's fair to say that installing almost anything from pypi for MicroPython will do it wrong.

Apart from the micropython-stubs that is :-)

**Reply by scruss** [reply]
> Apart from the micropython-stubs that is :-)

Well, technically: *yes*, but they aren't libraries that run under MicroPython
